src/SemiTorusStateSpace.cpp

```diff
@@ -0,0 +1,155 @@
+/***********************************************************************
+
+Copyright (c) 2015, Carnegie Mellon University
+All rights reserved.
+
+Authors: Chris Dellin <cdellin@gmail.com>
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+
+  Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*************************************************************************/
+
+#include <ompl/util/Exception.h>
+#include <ompl/base/spaces/RealVectorStateSpace.h>
+#include <or_ompl/SemiTorusStateSpace.h>
+
+or_ompl::SemiTorusStateSpace::SemiTorusStateSpace(unsigned int dim):
+   ompl::base::RealVectorStateSpace(dim),
+   isWrapping_(dim, false)
+{
+}
+
+void or_ompl::SemiTorusStateSpace::setIsWrapping(const std::vector<bool> &isWrapping)
+{
+   if (isWrapping_.size() != dimension_)
+      throw ompl::Exception("IsWrapping does not match dimension of state space: expected dimension " +
+         boost::lexical_cast<std::string>(dimension_) + " but got dimension " +
+         boost::lexical_cast<std::string>(isWrapping_.size()));
+   isWrapping_ = isWrapping;
+}
```

Comment:
What is the motivation behind making this a setter, instead of a constructor argument? I can't imagine wanting to modifying `isWrapping` after constructing the state space.